pipelines/pipedrive/custom_fields_munger.py

```diff
@@ -0,0 +1,85 @@
+from dlt.common.normalizers.names.snake_case import normalize_column_name
+from enum import Enum
+from typing import Dict, Generator, Tuple, Union
+
+import dlt
+import json
+import functools
+
+_custom_fields_mapping = {}
+
+
+class EndpointType(Enum):
+    ENTITY = 'ENTITY'
+    ENTITY_FIELDS = 'ENTITY_FIELDS'
+
+
+def get_munge_endpoint(endpoint: str = '', pipedrive_api_key: str = dlt.secrets.value, extra_params: Dict = None, endpoint_type: EndpointType = None) -> Generator[Dict, None, None]:
+    """
+    Generic function to retrieve and munge endpoint data.
+    """
+
+    from . import get_endpoint  # workaround
+
+    if all([endpoint, endpoint_type in EndpointType]):
+        data_pages = get_endpoint(endpoint, pipedrive_api_key, extra_params=extra_params)
+        munge_func = _munge_push_func if endpoint_type == EndpointType.ENTITY_FIELDS else _pull_munge_func
+        for data_page in data_pages:
+            data_page_map = map(functools.partial(munge_func, endpoint=endpoint), data_page)
+            yield from data_page_map
+
+
+def _munge_push_func(data_item: Dict = None, endpoint: str = '') -> Dict:
+    """
+    Specific function to munge data and push changes to custom fields' mapping
+    The endpoint must be an entity fields' endpoint
+    """
+    if all([data_item, data_item.get('edit_flag'), data_item.get('name'), data_item.get('key'), endpoint]):
+        normalized_name = data_item['name'].strip()  # remove leading and trailing spaces (e.g. 'MULTIPLE OPTIONS ' -> 'multiple_options_')
+        normalized_name = normalized_name.lower()  # typos' workaround (e.g. 'CUSTOMFIELD_DEaL_USER' -> 'customfield_d_ea_l_user')
```

Comment:
you can keep original names when renaming. dlt will normalize all those columns anyway. no need to do that

Reply:
I need to store normalized names in the mapping, in order to replace the hash strings with them in the entities and entity fields tables.
I'll remove the "typos' workaround" line but maybe it would be nice to document it somewhere.
Could it be possible to keep the "strip" line or maybe add it on dlt's "normalize_column_name" method?

Reply:
I still do not understand why you normalize the names here. You can replace the hashes with non normalized name and then `dlt` will do exactly the same thing you are doing here in normalization stage.

Where you should indeed normalize the names is in `get_parse_mappings` because we want to have a table of normalized names to hashes (honestly I'd include normalized name, non normalized name and hash in this table!)

btw. thanks for pointing our the spaces... I actually change them into "_". I will strip them as you suggest

Reply:
We've already discussed it, but in a nutshell, because dlt does normalize column names but it doesn't normalize cells' data.

Done!

My pleasure, great!